The configure script is over cautious about ANSI C support when cross-compiling.
Simply workaround this by disabling the test code.

diff -Naur configure.orig configure
--- configure.orig	2014-11-19 08:24:01.000000000 +0100
+++ configure	2014-11-20 17:51:11.301074527 +0100
@@ -4374,7 +4374,7 @@
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the compiler works on ANSI C" >&5
 $as_echo_n "checking whether the compiler works on ANSI C... " >&6; }
   if test "$cross_compiling" = yes; then :
-  as_fn_error $? "Couldn't build even a trivial ANSI C program: check CC." "$LINENO" 5
+  :
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
